greenheart/simulation/technologies/ammonia/ammonia.py

```diff
@@ -156,6 +156,70 @@ def run_ammonia_model(
 
     return ammonia_production_kgpy
 
+@define
+class AmmoniaSizeModelConfig:
+    """
+    Configuration inputs for the ammonia sizing model, including plant capacity and
+    feedstock details.
+
+    Attributes:
+        hydrogen_amount_kg Optional (float): The amount of hydrogen available in kilograms 
+            per year to make ammonia.
+        ammonia_plant_size_kgpy Optional (float): The amount of desired ammonia production in
+            kilograms per year.
+        plant_capcity_factor (float): The ammonia plant capacity factor.
+        feedstocks (Feedstocks): An instance of the `Feedstocks` class detailing the
+            costs and consumption rates of resources used in production.
+    """
+    plant_capacity_factor: float
+    feedstocks: Feedstocks
+    hydrogen_amount_kg: Optional[float] = field(default=None)
+    ammonia_plant_size_kgpy: Optional[float] = field(default=None)
+
+
+    def __attrs_post_init__(self):
+        if self.hydrogen_amount_kg is None and self.ammonia_plant_size_kgpy is None:
+            raise ValueError("`hydrogen_amount_kg` or `ammonia_plant_size_kgpy` is a required input.")
+
+        if self.hydrogen_amount_kg and self.ammonia_plant_size_kgpy:
+            raise ValueError("can only select one input: `hydrogen_amount_kg` or `ammonia_plant_size_kgpy`.")
+
+@define
+class AmmoniaSizeModelOutputs:
+    """
+    Outputs from the ammonia size model.
+
+    Attributes:
+        ammonia_plant_size_kgpy (float): If amount of hydrogen in kilograms per year is input, 
+            the size of the ammonia plant in kilograms per year is output.
+        hydrogen_amount_kg (float): If amount of ammonia production in kilograms per year is input, 
+            the amount of necessary hydrogen feedstock in kilograms per year is output.
+    """
+    ammonia_plant_size_kgpy: float
+    hydrogen_amount_kg: float
+
+def run_size_ammonia_plant(config: AmmoniaSizeModelConfig) -> AmmoniaSizeModelOutputs:
+    
+    if config.hydrogen_amount_kg:
+        ammonia_plant_size_kgpy = (config.hydrogen_amount_kg 
+            / config.feedstocks.hydrogen_consumption 
+            * config.plant_capacity_factor
+        )
+        hydrogen_amount_kg = config.hydrogen_amount_kg
+
+    if config.ammonia_plant_size_kgpy:
+        hydrogen_amount_kg = (config.ammonia_plant_size_kgpy 
+            * config.feedstocks.hydrogen_consumption
+            / config.plant_capacity_factor
```

Comment:
We should probably call this something like `input_capacity_factor_estimate` or something so no one takes it as an output.

Reply:
We need to be clear about what capacity factor we are estimating here. Is this the capacity factor of the ammonia plant or the energy plant?

Reply:
I think your if statements here may be flipped. If the user provides the hydrogen amount, the sizing should return the ammonia amount, and vice versa.